backend/app/resources.py

```diff
@@ -94,9 +103,86 @@ class SubcategoryList(Resource):
     def get(self):
         return marshal(Subcategory.query.all(), subcategory_marshaller)
 
+    @jwt_required
     def post(self):
         args = creating_subcategory_parser.parse_args()
-        subcategory = Repository.create_and_add(args)
+        subcategory = Repository.create_and_add(Subcategory, args)
         if not subcategory:
-            return 404
+            return {'message': 'Something went wrong'}, 500
         return marshal(subcategory, subcategory_marshaller), 201
+
+
+class UserRegistration(Resource):
+    def post(self):
+        args = user_parser.parse_args()
+
+        try:
+            new_user = User(username=args['username'])
+            db.session.add(new_user)
+            new_user.password_hash = args['password']
```

Comment:
This `try/except` block does far too much. Split it into smaller methods or something.